src/main.c

```diff
@@ -351,18 +362,120 @@ key_binding_func_key (MBWindowManager   *wm,
   hd_dbus_send_event (s);
 }
 
+/* Close all input devices previously opened */
+static void
+close_input_devices (Display *dpy)
+{
+  int i;
+
+  for (i = 0; i < xi_devices->len; i++)
+    {
+      hd_xi_device *xi_dev = &g_array_index (xi_devices, hd_xi_device, i);
+
+      if (xi_dev->dev)
+        {
+          XCloseDevice (dpy, xi_dev->dev);
+          xi_dev->dev = NULL;
```

Comment:
This is not needed.

Reply:
because it will prevent SEGFAULTs if somebody decides to (wrongly) call the function for the second time. Also, we better have no dangling pointers around.